src/Integration.Vsix/Settings/GeneralOptionsDialogPage.cs

```diff
@@ -33,7 +34,19 @@ internal class GeneralOptionsDialogPage : UIElementDialogPage
         private GeneralOptionsDialogControl dialogControl;
         private ISonarLintSettings settings;
 
-        protected override UIElement Child => dialogControl ?? (dialogControl = new GeneralOptionsDialogControl());
+        protected override UIElement Child
+        {
+            get
+            {
+                if (dialogControl == null)
+                {
+                    Debug.Assert(this.Site != null, "Expecting the page to be sited");
```

Comment:
The page has access to a service provider via the _Site_ property. It is now responsible for fetching the daemon etc required by the child control